### PR TITLE
minor fixes of the workflow to compile

### PR DESCRIPTION
**Pull request by gretel**
- trigger on all 'push' and 'pull_request' events
- fix workflow 'name' having spaces (used for artifact filename)
- use GITHUB_REF_NAME to include branch/pr id in artifact filename

### CODE REVIEW

File: .github/workflows/compile.yml [hunk 1]
```diff
@@ -38,7 +34,7 @@ jobs:
     - name: Compile All
       run: |
         ./make
-        echo "ARTIFACT_NAME=${GITHUB_WORKFLOW}-$(git rev-parse --short ${GITHUB_SHA})-${GITHUB_RUN_NUMBER}" >> "${GITHUB_ENV}"
+        echo "ARTIFACT_NAME=${GITHUB_WORKFLOW}-${GITHUB_REF_NAME}-$(git rev-parse --short ${GITHUB_SHA})-${GITHUB_RUN_NUMBER}" >> "${GITHUB_ENV}"
```
Comment (johnou) [inline .github/workflows/compile.yml]: @gretel see https://github.com/ToyKeeper/anduril/pull/6#discussion_r1382676058

Comment (johnou) [inline .github/workflows/compile.yml]:
```suggestion
        SAFE_GITHUB_REF_NAME=$(echo "${GITHUB_REF_NAME}" | tr '":<>|*?\\/' '_')
        echo "ARTIFACT_NAME=${GITHUB_WORKFLOW}-${SAFE_GITHUB_REF_NAME}-$(git rev-parse --short ${GITHUB_SHA})-${GITHUB_RUN_NUMBER}" >> "${GITHUB_ENV}"
```

**Review by johnou — CHANGES_REQUESTED** [review]

File: .github/workflows/compile.yml (hunk 1 above)
Comment (gretel) [reply .github/workflows/compile.yml]: yea that's better

**Review by johnou — APPROVED** [review]

**Comment by ToyKeeper** [comment]
Running on all push/pull_request events seems like a good idea.  It doesn't seem to need the ref name to guarantee a unique filename though, since it already has the revision hash and run number.  Is there a reason that field would be needed?

**Comment by gretel** [comment]
> Running on all push/pull_request events seems like a good idea. It doesn't seem to need the ref name to guarantee a unique filename though, since it already has the revision hash and run number. Is there a reason that field would be needed?

it's not a requirement.. just useful for reference; the downloaded file(name) has some basic semantics to figure back on the action it was resulting from.

**Comment by ToyKeeper** [comment]
Not really sure it needs the ref name in the file name.  It has pros and cons though:

- Pro: Makes collections of artifact files a bit easier to sort through.  But I'm not aware of anyone in this project who actually does that, so I don't know of anyone who would see any benefits.
- Con: Makes the process more brittle and prone to failing if users put anything unexpected in a text field.

So, closing for now unless there's a compelling reason to expand the filenames.

**Pull request closed without merging**